### PR TITLE
Synchronize rhcos artifacts from ART to CI

### PR DESCRIPTION
**Pull request by jupierce**
There are a messy set of prow based periodic promotion jobs which handled promotion of machine-os-content images from their staging area after a build into the CI imagestreams. The purpose of these jobs was to run an e2e test on the jobs before promoting into CI.
With 4.12, there is significantly more pre-merge testing on RHCOS and there are a growing number of images that make the old promotion jobs difficult to maintain. 
This PR moves the responsibility for OCP 4.12 and onward into ART's pipeline.

### CODE REVIEW

File: jobs/build/build-sync/build.groovy [hunk 1]
```diff
@@ -90,6 +90,49 @@ ${excludeArchesParam}
...
+            // references to the latest machine-os-content, rhel-coreos-8, rhel-coreos-8-extensions (and
+            // potentially more with rhel9). If this is failing, it must be treated as a priority since
+            // CI will begin falling being nightly CoreOS content.
+            if ( major > 4 || minor >= 12 ) {
+                def arches = buildlib.branch_arches(groupParam)
+                // Gather a list of tags to mirror to the CI imagestream. This will include rhel-coreos-* to pick up
+                // any future RHCOS RHEL versions (e.g. rhel-coreos-9).
+                def tags = commonlib.shell(
+                    script: """
+                    oc --kubeconfig ${buildlib.ciKubeconfig} get -n ocp is/${params.BUILD_VERSION}-art-latest -o=json | jq .spec.tags[].name -r | grep -e machine-os-content -e rhel-coreos-
```
Comment (cgwalters) [inline jobs/build/build-sync/build.groovy:109]: We will want a more principled thing than `grep` here in the future - this relates to things like https://github.com/openshift/doozer/blob/3e755229ca4e00c4fd34e968a56518102ab32b6b/doozerlib/rhcos.py#L20

too.

But, seems OK to me for now.

File: jobs/build/build-sync/build.groovy [hunk 2]
```diff
@@ -90,6 +90,49 @@ ${excludeArchesParam}
...
+    if (params.ASSEMBLY == "stream" && !params.DOOZER_DATA_GITREF) {
+        slackChannel = slacklib.to(params.VERSION)
+        slackChannel.task("Synchronizing ART CoreOS with CI for ${params.VERSION}") {
+            buildlib.oc("${logLevel} --kubeconfig ${buildlib.ciKubeconfig} registry login")
+            def (major, minor) = commonlib.extractMajorMinorVersionNumbers(params.BUILD_VERSION)
+            // Starting with 4.12, ART is responsible for populating the CI imagestream (-n ocp is/4.12) with
+            // references to the latest machine-os-content, rhel-coreos-8, rhel-coreos-8-extensions (and
+            // potentially more with rhel9). If this is failing, it must be treated as a priority since
+            // CI will begin falling being nightly CoreOS content.
+            if ( major > 4 || minor >= 12 ) {
```
Comment (cgwalters) [inline jobs/build/build-sync/build.groovy:103]: Shouldn't this be `major >= 4`?

Comment (cgwalters) [reply jobs/build/build-sync/build.groovy:103]: This looks like the only major issue with this PR.

Comment (jupierce) [reply jobs/build/build-sync/build.groovy:103]: The >= 12 captures 4.12 and beyond.

Comment (cgwalters) [reply jobs/build/build-sync/build.groovy:103]: Ah right sorry yep

**Pull request merged**